### PR TITLE
[ci] Add dockerfile.ubuntu-20.04.amdgpu

### PR DESCRIPTION
**Pull request by GaleSeLee**
Issue: https://github.com/taichi-dev/taichi/issues/6434

### Brief Summary
This dockerfile refs to Dockerfile.ubuntu.20.04. The differences are following
1. based docker image
2. del python3-pip, mesa-common-dev which are already downloaded in based image.
3. remove vulkan module(amdgpu is not yet adapted to vulkan)
4. add user into video usergroup. (Some Ubuntu-20.04 versions also require render group)

### CODE REVIEW

File: ci/Dockerfile.ubuntu.20.04.amdgpu [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+                       libwayland-dev \
+                       bison \
+                       libx11-xcb-dev \
+                       liblz4-dev \
+                       libzstd-dev \
+                       qt5-default \
+                       libglfw3 \
+                       libglfw3-dev \
+                       libjpeg-dev
+# Install the latest version of CMAKE v3.20.5 from source
```
Comment (feisuzhu) [inline ci/Dockerfile.ubuntu.20.04.amdgpu:38]: cmake is now installed by `pip`, so we can remove this block

File: ci/Dockerfile.ubuntu.20.04.amdgpu [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+WORKDIR /
+RUN wget https://github.com/Kitware/CMake/releases/download/v3.20.5/cmake-3.20.5-linux-x86_64.tar.gz
+RUN tar xf cmake-3.20.5-linux-x86_64.tar.gz && \
+    rm cmake-3.20.5-linux-x86_64.tar.gz
+ENV PATH="/cmake-3.20.5-linux-x86_64/bin:$PATH"
+
+# Install LLVM 15
+WORKDIR /
+# Make sure this URL gets updated each time there is a new prebuilt bin release
+RUN wget https://github.com/GaleSeLee/assets/releases/download/v0.0.1/taichi-llvm-15.0.0-linux.zip
```
Comment (feisuzhu) [inline ci/Dockerfile.ubuntu.20.04.amdgpu:48]: Please use @ailzhang 's version, grep `taichi-llvm` from `.github` directory to find the link.

Comment (GaleSeLee) [reply ci/Dockerfile.ubuntu.20.04.amdgpu:41]: We use a different target of llvm. This is for `x86` and `amdgpu` while `taichi-llvm` from `.github` is for `x86` and `nvptx`

**Review by feisuzhu — APPROVED** [review]
LGDM

**Pull request merged**